packages/native/src/components/Form/Input/BaseInput/index.tsx

```diff
@@ -83,9 +104,33 @@ export const InputRenderRightContainer = styled(FlexBox).attrs(() => ({
   pr: "16px",
 }))``;
 
-export default function Input(props: InputProps): JSX.Element {
-  const { value, disabled, error, renderLeft, renderRight, ...textInputProps } =
-    props;
+// Yes, this is dirty. If you can figure out a better way please change the code :).
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const IDENTITY = (_: any): any => _;
+
+export default function Input<T = string>(props: InputProps<T>): JSX.Element {
+  const {
+    value,
+    onChange,
+    onChangeText,
+    disabled,
+    error,
+    renderLeft,
+    renderRight,
+    serialize = IDENTITY,
+    deserialize = IDENTITY,
+    ...textInputProps
+  } = props;
+
+  const inputValue = useMemo(() => serialize(value), [serialize, value]);
+
+  const handleChange = useCallback(
+    (value: string) => {
+      onChange(deserialize(value));
+      if (onChangeText) onChangeText(value);
```

Comment:
We could  be able to use onChangeText without having onChange mandatory ?

Reply:
> We should be able to use onChangeText without having onChange mandatory ?

True! I also added an `onChangeEvent` prop to be able to use the native event - and made `onChange` completely optional.

Reply:
Nice !

I was thinking about the onChangeEvent. For desktop and native input, Shouldn't we keep the onChange event raw, and use a special prop for this enhanced onChange, rather than doing the opposite ? A little like what native is doing with the onChangeText event.

Reply:
> Shouldn't we keep the onChange event raw, and use a special prop for this enhanced onChange, rather than doing the opposite

It samkes perfect sense, but from my experience it is more common to re-use `onChange` (this is also the current LL behaviour).